### PR TITLE
[clang][analyzer][doc] Migrate checkers-related docs from HTML to RST

### PR DESCRIPTION
**Pull request by gamesh411**
Documentation for the checkers is kept up to date in RST files.
This patch removes duplication by replacing the HTML docs with links to
docs generated from the RST.

### CODE REVIEW

**Comment by gamesh411** [comment]
I have tried to ensure no information is lost with this change.
I have cross-checked the history of the checker's HTML files with commits that introduce checkers and have found that we tend to add checkers only to the RST files and sometimes miss the fact that there is another description in the HTML format.

**Review by steakhal — APPROVED** [review]
LGTM, thanks.

**Comment by NagyDonat** [comment]
:thinking: Deleting the html files could break some links on external sites, so I think it would be better to replace them with a very simple "This content was moved to <link>" placeholder.

File: clang/www/analyzer/checker_dev_manual.html [hunk 1]
```diff
@@ -678,7 +678,7 @@ <h3 id=debugprints>Debug Prints and Tricks</h3>
 
 <h2 id=links>Making Your Checker Better</h2>
 <ul>
-<li>User facing documentation is important for adoption! Make sure the <a href="/available_checks.html">checker list </a>is updated
+<li>User facing documentation is important for adoption! Make sure the <a href="https://clang.llvm.org/docs/analyzer/checkers.html">checker list </a>is updated
```
Comment (whisperity) [inline clang/www/analyzer/checker_dev_manual.html:681]:
```suggestion
<li>User facing documentation is important for adoption! Make sure the <a href="https://clang.llvm.org/docs/analyzer/checkers.html">checker list</a> is updated
```

File: clang/www/analyzer/annotations.html [hunk 2]
```diff
@@ -126,7 +126,7 @@ <h3 id="cocoa_mem">Cocoa &amp; Core Foundation Memory Management
 Annotations</h3>
 
 <!--
-<p>As described in <a href="/available_checks.html#retain_release">Available
+<p>As described in <a href="https://clang.llvm.org/docs/analyzer/checkers.html">Available
 Checks</a>,
 -->
```
Comment (whisperity) [inline clang/www/analyzer/annotations.html:131]: This is just a comment, I think we can remove this completely. Also, the original URL contained an anchor tag, but the new one does not, so the two sides aren't actually equivalent.

**Comment by whisperity** [comment]
> > 🤔 Deleting the html files could break some links on external sites, so I think it would be better to replace them with a very simple "This content was moved to `<link>`" placeholder.
> 
> Maybe something like this could help: https://stackoverflow.com/questions/5411538/how-to-redirect-one-html-page-to-another-on-load

@gamesh411 Yes, please. :slightly_smiling_face: I think we could do it in a way that we say that from 19.0 to 21.0 or something, the redirects (and the clickable URL) is there, and then remove everything completely later. Just put up a warning that we are redirecting now but the links will break!

(FYI, Tidy check aliases are also implemented via such redirect pages.)

File: clang/www/analyzer/annotations.html (hunk 2 above)
Comment (gamesh411) [reply clang/www/analyzer/annotations.html]: I am unsure which one to link to, so I opted for less but not misleading information with this edit.

**Pull request merged**